modyn/supervisor/internal/grpc_handler.py

```diff
@@ -434,3 +462,173 @@ def seed_selector(self, seed: int) -> None:
         ).success
 
         assert success, "Something went wrong while seeding the selector"
+
+    def start_evaluation(self, trained_model_id: int, pipeline_config: dict) -> dict[int, EvaluationStatusTracker]:
+        if not self.connected_to_evaluator:
+            raise ConnectionError("Tried to start evaluation at evaluator, but there is no gRPC connection.")
+
+        model_id = pipeline_config["model"]["id"]
+        if "config" in pipeline_config["model"]:
+            model_config = json.dumps(pipeline_config["model"]["config"])
+        else:
+            model_config = "{}"
+
+        device = pipeline_config["evaluation"]["device"]
+        amp = pipeline_config["evaluation"]["amp"] if "amp" in pipeline_config["evaluation"] else False
+
+        evaluations: dict[int, EvaluationStatusTracker] = {}
+
+        # enforce unique dataset_ids
+        dataset_ids_seen = set()
+        for dataset in pipeline_config["evaluation"]["datasets"]:
+            dataset_id = dataset["dataset_id"]
+            if dataset_id in dataset_ids_seen:
+                raise ValueError("Dataset id should be unique among evaluated datasets.")
+            dataset_ids_seen.add(dataset_id)
+            if "transformations" in dataset:
+                transform_list = dataset["transformations"]
+            else:
+                transform_list = []
+
+            if "label_transformer_function" in dataset:
+                label_transformer = dataset["label_transformer_function"]
+            else:
+                label_transformer = ""
+
+            bytes_parser_function = dataset["bytes_parser_function"]
+            batch_size = dataset["batch_size"]
+            dataloader_workers = dataset["dataloader_workers"]
```

Comment:
I think there might be an indentation problem here. Should this be within the for loop for the datasets?

Reply:
Yes, I would say that `batch_size` and `dataloader_workers` are dataset-dependent. However, I can change this if it doesn't make sense.

Reply:
Ahh you're right, sorry I am not sure what I mixed up here.